src/psyclone/tests/dynamo0p3_test.py

```diff
@@ -3318,8 +3283,8 @@ def test_halo_ex_back_dep_no_call(monkeypatch):
     # being of type 'function'. I'm not sure why. However, this does
     # not matter in practice as we are just trying to get PSyclone to
     # raise the appropriate exception.
-    assert ("Generation Error: In HaloInfo class, field 'f2' should be from a "
-            "call but found %s" % type(lambda: halo_exchange)
+    assert (f"Generation Error: In HaloInfo class, field 'f2' should be from "
+            f"a call but found {str(type(lambda: halo_exchange))}"
```

Comment:
Since we're here, would you mind replacing L3277 with just `halo_exchange)` (i.e. remove the use of `lambda`) and see whether than then means we can remove the `lambda` here. If so, please update the comment too.

Reply:
We can remove `lambda` so I tidied up the test. I removed the comment alltogether and I also removed the added `str`.

Reply:
Great.